### PR TITLE
operator: Support TLS enabled lokistack-gateway (Kubernetes native)

### PR DESCRIPTION
**Pull request by aminesnow**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [X] Tests updated
- [x] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: operator/internal/manifests/ingester.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 // BuildIngester builds the k8s objects required to run Loki Ingester
 func BuildIngester(opts Options) ([]client.Object, error) {
 	statefulSet := NewIngesterStatefulSet(opts)
-	if opts.Flags.EnableTLSServiceMonitorConfig {
+	if opts.Flags.EnableHTTPTLSServices {
 		if err := configureIngesterServiceMonitorPKI(statefulSet, opts.Name); err != nil {
```
Comment (periklis) [inline operator/internal/manifests/ingester.go:27]: dito here like with the compactor.

File: operator/internal/manifests/query-frontend.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ import (
 // BuildQueryFrontend returns a list of k8s objects for Loki QueryFrontend
 func BuildQueryFrontend(opts Options) ([]client.Object, error) {
 	deployment := NewQueryFrontendDeployment(opts)
-	if opts.Flags.EnableTLSServiceMonitorConfig {
+	if opts.Flags.EnableHTTPTLSServices {
 		if err := configureQueryFrontendServiceMonitorPKI(deployment, opts.Name); err != nil {
```
Comment (periklis) [inline operator/internal/manifests/query-frontend.go:21]: dito here like with the compactor.

File: operator/internal/manifests/ruler.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ import (
 // BuildRuler returns a list of k8s objects for Loki Stack Ruler
 func BuildRuler(opts Options) ([]client.Object, error) {
 	statefulSet := NewRulerStatefulSet(opts)
-	if opts.Flags.EnableTLSServiceMonitorConfig {
+	if opts.Flags.EnableHTTPTLSServices {
 		if err := configureRulerServiceMonitorPKI(statefulSet, opts.Name); err != nil {
```
Comment (periklis) [inline operator/internal/manifests/ruler.go:24]: dito here like with the compactor.

File: operator/internal/manifests/service_monitor.go [hunk 4]
```diff
@@ -148,7 +148,7 @@ func newServiceMonitor(namespace, serviceMonitorName string, labels labels.Set,
 	}
 }
 
-func configureServiceMonitorPKI(podSpec *corev1.PodSpec, serviceName string) error {
+func configureTLS(podSpec *corev1.PodSpec, serviceName string) error {
```
Comment (periklis) [inline operator/internal/manifests/service_monitor.go:151]: I believe we need to move this function out of `service_monitor.go`

File: operator/bundle/manifests/loki-operator-manager-config_v1_configmap.yaml [hunk 5]
```diff
@@ -18,6 +18,7 @@ data:
       enableCertSigningService: true
       enableServiceMonitors: true
       enableTlsServiceMonitorConfig: true
+      enableHttpTLSServices: true
       enableTlsGRPCServices: true
```
[the remark this answers is not in the document]
Comment (aminesnow) [reply operator/bundle/manifests/loki-operator-manager-config_v1_configmap.yaml:22]: I don't understand what you mean by this..

File: operator/internal/manifests/service_monitor.go (hunk 4 above)
Comment (aminesnow) [reply operator/internal/manifests/service_monitor.go:151]: Sure. Was thinking abt moving it to `config.go`, WDYT?

File: operator/bundle/manifests/loki-operator-manager-config_v1_configmap.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (periklis) [reply operator/bundle/manifests/loki-operator-manager-config_v1_configmap.yaml:22]: @aminesnow I mean that we have two ways to name things, e.g. `enableHttpTlsServices` vs. `enableTlsGrpcServices`, either we pad `Tls` after Http/Grpc or before.

**Review by periklis — COMMENTED** [review]
Looks good to me besides the feature flag naming issue.

File: operator/bundle/manifests/loki-operator-manager-config_v1_configmap.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (aminesnow) [reply operator/bundle/manifests/loki-operator-manager-config_v1_configmap.yaml:22]: Oh I see!

**Review by periklis — APPROVED** [review]

**Pull request merged**